src/xdg.c

```diff
@@ -17,6 +17,23 @@
 #define LAB_XDG_SHELL_VERSION (2)
 #define CONFIGURE_TIMEOUT_MS 100
 
+static bool
+xdg_toplevel_view_contains_window_type(struct view *view, int32_t window_type)
+{
+	assert(view);
+	assert(view->type == LAB_XDG_SHELL_VIEW);
```

Comment:
The 2nd assert is not necessary, the abstraction and the function being static already makes sure that its always a xdg view.